### PR TITLE
Feature/#13 votings api

### PR DESCRIPTION
**Pull request by pedalopon**
## changes
- **Get votings endpoint**

  Descripción: Devuelve todas las votaciones
  GET http://localhost:8000/administration/api/votings
  Ningún atributo obligatorio
- **Create voting endpoint** 

  Descripción: Crea una nueva votacion
  POST http://localhost:8000/administration/api/votings
  Atributos obligatorios body: Objeto voting tal y como se especifica en la tarea #13 
  EJ:
```json
  {
      "name": "Mejor asignatura",
      "desc": "vota tu asignatura favorita",
      "question": {
			  "desc": "describiendo",
				  "options": [
					  {
						  "number": 1,
						  "option": "dp"
					  },
					  {
						  "number": 2,
						  "option": "adda"
					  }
				  ]
		  },
      "auth": "http://localhost:8000",
      "census": [1,2]
  }
```
- **Voting actions endpoint**

  Descripción: Aplica la accion seleccionada a las votaciones que se han pasado como id.
  PUT http://localhost:8000/administration/api/votings
  Atributos obligatorios body: idList(Lista de id de votaciones), action(String que solo puede tomar los valores: start, stop y tally)
  EJ:
```json
  {
	  "idList":[ 16 ],
	  "action":"tally"
  }
```
- **BulkDelete voting endpoint**

  Descripción: Se eliminan las votaciones que corresponden a ese id, sino se pasa ninguna se borran todas
  DELETE http://localhost:8000/administration/api/votings
  Atributos obligatorios en el Body: idList(Lista de id de votaciones)
  EJ:
```json
  {
	  "idList":[ 12,16,17]
  }
```
## Test
- Realizar una petición a cada endpoint pasando sus atributos obligatorios y comprobando su correcto funcionamiento

### CODE REVIEW

File: decide/administration/serializers.py [hunk 1]
```diff
@@ -29,10 +32,11 @@ class UserUpdateSerializer(serializers.Serializer):
 
 class AdminQuestionSerializer(serializers.ModelSerializer):
     options = AdminQuestionOptionSerializer(many=True)
+
     class Meta:
         model = Question
-        fields = ('id','desc', 'options')
-        depht=1
+        fields = ('id', 'desc', 'options')
+        depht = 1
```
Comment (JSnow11) [inline decide/administration/serializers.py]: this is misspelt, I guess you mean **_depth_**

File: decide/administration/views.py [hunk 2]
```diff
@@ -20,6 +25,115 @@ def index(request):
...
+            is_valid(len(ids) > 0, 'The format of the ids list is not correct')
+            Voting.objects.filter(id__in=ids).delete()
+            return Response({}, status=HTTP_200_OK)
+
+class VotingsAPI(APIView):
+    permission_classes = (IsAdminAPI,)
+
+    def get(self, request, voting_id):
+        try:
+            query = Voting.objects.filter(id=voting_id).get()
```
Comment (JSnow11) [inline decide/administration/views.py]: why is this called query?, maybe **_voting_** or **_result_**

File: decide/administration/views.py [hunk 3]
```diff
@@ -20,6 +25,115 @@ def index(request):
     return render(request, "build/index.html")
 
 
+class VotingAPI(APIView):
+    permission_classes = (IsAdminAPI,)
+
+    def get(self, request):
+        query = Voting.objects.all()
```
Comment (JSnow11) [inline decide/administration/views.py]: why query? maybe _**results**_ or _**votings**_?

**Review by JSnow11 — CHANGES_REQUESTED** [review]
Added some comments about small things with naming and variables

File: decide/administration/serializers.py (hunk 1 above)
Comment (pedalopon) [reply decide/administration/serializers.py]: Typo fixed in last commit

File: decide/administration/views.py (hunk 2 above)
Comment (pedalopon) [reply decide/administration/views.py]: Query is also fine like it is used in djiango wiki for db operations but for readability has been changed in last commit.

File: decide/administration/views.py (hunk 3 above)
Comment (pedalopon) [reply decide/administration/views.py]: Query is also fine like it is used in djiango wiki for db operations but for readability has been changed in last commit.

File: decide/administration/serializers.py (hunk 1 above)
Comment (JSnow11) [reply decide/administration/serializers.py]: still being _depht_ instead of **_depth_** @pedalopon

**Review by JSnow11 — APPROVED** [review]
All good

**Pull request merged**